crates/tabby/src/services/code.rs

```diff
@@ -148,21 +147,11 @@ impl CodeSearch for CodeSearchImpl {
     }
 }
 
-fn strip_url_authentication(url: &str) -> String {
-    Url::parse(url)
-        .map(|mut url| {
-            let _ = url.set_password(None);
-            let _ = url.set_username("");
-            url.to_string()
-        })
-        .unwrap_or_else(|_| url.to_string())
-}
-
 fn closest_match<'a>(
     search_term: &'a str,
     search_input: impl IntoIterator<Item = &'a str>,
```

Comment:
```suggestion
    search_input: impl IntoIterator<Item = &'a RepositoryConfig>,
```

passin repository config directly and invoke `canonical_git_url`?